plugin/trino-faker/src/main/java/io/trino/plugin/faker/FakerPageSource.java

```diff
@@ -518,7 +567,18 @@ private ObjectWriter objectWriter(Type type)
         if (type instanceof CharType charType) {
             return (blockBuilder, value) -> charType.writeSlice(blockBuilder, (Slice) value);
         }
-        // not supported: ROW, ARRAY, MAP, JSON
+        if (type instanceof RowType rowType) {
+            return (blockBuilder, value) -> rowType.writeSlice(blockBuilder, (Slice) value);
+        }
+        if (type instanceof ArrayType arrayType) {
+            return (blockBuilder, value) -> arrayType.writeSlice(blockBuilder, (Slice) value);
```

Comment:
This might be wrong, I think we're missing test coverage here. I need to add these types in `testSelectValuesProperty`